full-service/src/service/sync.rs

```diff
@@ -311,100 +311,96 @@ fn sync_thread_entry_point(
     }
 }
 
-/// Sync a single account.
-pub fn sync_account(
+/// Sync a single account against the next chuck of blocks.
+pub fn incrementally_sync_account(
     ledger_db: &LedgerDB,
     wallet_db: &WalletDb,
     account_id: &str,
     logger: &Logger,
-) -> Result<SyncAccountOk, SyncError> {
-    for _ in 0..MAX_BLOCKS_PROCESSING_CHUNK_SIZE {
-        let conn = wallet_db.get_conn()?;
-        let sync_status = conn.transaction::<SyncAccountOk, SyncError, _>(|| {
-            // Get the account data. If it is no longer available, the account has been
-            // removed and we can simply return.
+) -> Result<AccountSyncStatus, SyncError> {
+    // Index of the last block currently in the ledger.
+    let final_block_index = ledger_db.num_blocks()? - 1;
+
+    let conn = wallet_db.get_conn()?;
```

Comment:
Is there a reason that `conn` was opened inside the for loop? I **think** it's safe to get the connection once, and then create a separate DB transaction on each iteration of the loop...

Reply:
That depends on the pool size - if the size is small (e.g. 1 - https://github.com/mobilecoinofficial/full-service/blob/main/full-service/src/db/wallet_db.rs#L22) that means that until the loop finished executing, nothing else gets to get a connection.

Reply:
Thanks. I will leave it in the loop